viz/charts/linkerd-viz/templates/metrics-api.yaml

```diff
@@ -72,8 +72,8 @@ spec:
         - -cluster-domain={{.Values.clusterDomain}}
         {{- if .Values.prometheusUrl }}
         - -prometheus-url={{.Values.prometheusUrl}}
-        {{- else }}
-        - -prometheus-url=http://linkerd-prometheus.linkerd-viz.svc.{{.Values.clusterDomain}}:9090
+        {{- else if .Values.prometheus.enabled }}
```

Comment:
What's the purpose of this change? It looks like the intent here is to skip rendering the `-prometheus-url` flag if both `.Values.prometheusUrl` is unset and `.Values.prometheus.enabled` is false.  But I think we should treat that as an invalid configuration.  Perhaps we just need an else block here that aborts with an error saying that prometheus.enabled or prometheusUrl must be set.

Reply:
This was important because when external prometheus is not set, and default is disabled we would not want the `linkerd viz check` to just run for sometime and fail. This makes it error out that there is no prometheus instance to connect.

> Perhaps we just need an else block here that aborts with an error saying that prometheus.enabled or prometheusUrl must be set.

Yep, That is the plan. I was hoping to address these Prometheus related changes in the follow-up PR.

Reply:
Cool.  Let's either do the proper thing here, or revert this change and handle it in the follow-up.

Reply:
Addressed the else case here itself